custom_model_runner/datarobot_drum/drum/model_adapter.py

```diff
@@ -338,11 +342,45 @@ def predict(self, input_filename, model=None, **kwargs):
                     "Model post-process hook failed to post-process predictions: {}".format(exc)
                 ).with_traceback(sys.exc_info()[2]) from None
 
-        if kwargs[TARGET_TYPE_ARG_KEYWORD] == TargetType.UNSTRUCTURED:
-            PythonModelAdapter._validate_unstructured_predictions(predictions)
+        self._validate_predictions(predictions, positive_class_label, negative_class_label)
+
+        return predictions
+
+    @staticmethod
+    def _validate_unstructured_predictions(unstructured_response):
+        def _is_instance_str_bytes_none(data):
+            return isinstance(data, (bytes, str, type(None)))
+
+        def _is_instance_dict_none(data):
+            return isinstance(data, (dict, type(None)))
+
+        if _is_instance_str_bytes_none(unstructured_response):
+            pass
+        elif isinstance(unstructured_response, tuple):
+            if len(unstructured_response) != 2:
+                raise ValueError(
+                    "In unstructured mode predictions of type tuple must have length = 2, got {}".format(
+                        len(unstructured_response)
+                    )
+                )
+            par1 = unstructured_response[0]
+            par2 = unstructured_response[1]
+            if not _is_instance_str_bytes_none(par1) or not _is_instance_dict_none(par2):
+                raise ValueError(
+                    "In unstructured mode predictions tuple must be of type (str, dict) but received ({}, {})".format(
```

Comment:
code says that message should say "str/bytes, dict"

Reply:
This error is shown when tuple (str, int) is returned.
So I mean if you try to return tuple, it has to be (str, dict) tuple

Reply:
***Fixed as you suggested***